File: src/main/java/com/evieclient/utils/saving/OtherWay/Save.java
package com.evieclient.utils.saving.OtherWay;

import com.evieclient.Evie;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import com.google.gson.stream.JsonWriter;
import io.sentry.Sentry;

import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

/**
 * Saving booleans to a json (and other things prob)
 *
 * @author Pinkulu | pinkulu#6260 *
 */
public class Save {

  /** run this whenever a change is made * */
  public static void saveConfig() {
    try {
      File file = new File(Evie.evieDir, "toggle.json");
      if (!file.exists()) {
        file.getParentFile().mkdirs();
        file.createNewFile();
      }
      JsonWriter writer = new JsonWriter(new FileWriter(file, false));
      writeJson(writer);
      writer.close();
    } catch (Throwable e) {
  Sentry.captureException(e);
      e.printStackTrace();
    }
  }

  /** run this when the game is Initializing * */
  public static void loadConfig() {
    try {
      File file = new File(Evie.evieDir, "toggle.json");
      if (file.exists()) readJson(file);
    } catch (Throwable e) {
  Sentry.captureException(e);
      e.printStackTrace();
    }
  }

  /** add the booleans from example * */
  public static void readJson(File file) throws Throwable {
    JsonParser parser = new JsonParser();
    JsonObject json = parser.parse(new FileReader(file)).getAsJsonObject();
    json = json.getAsJsonObject("Toggle");
  }

  /** add the booleans from example * */
  public static void writeJson(JsonWriter writer) throws IOException {
    writer.setIndent(" ");
    writer.beginObject();
    writer.name("First thing");
    writer.beginObject();
    writer.endObject();
    writer.endObject();
  }
}

File: src/main/java/com/evieclient/utils/saving/Save.kt
package com.evieclient.utils.saving

import com.evieclient.Evie
import com.evieclient.modules.hud.RenderModule
import com.google.gson.JsonObject
import java.io.FileWriter
import com.google.gson.JsonParser
import com.google.gson.stream.JsonWriter
import io.sentry.Sentry
import java.io.File
import java.io.FileReader
import java.io.IOException

object Save {

    fun saveConfig() {
        try {
            val file = File(Evie.evieDir, "toggle.json")
            if (!file.exists()) {
                file.parentFile.mkdirs()
                file.createNewFile()
            }
            val writer = JsonWriter(FileWriter(file, false))
            writeJson(writer)
            writer.close()
        } catch (e: Throwable) {
            Sentry.captureException(e)
            e.printStackTrace()
        }
    }

    fun loadConfig(): JsonObject? {
        try {
            val file = File(Evie.evieDir, "toggle.json")
            if (file.exists()) {
                return readJson(file);
            }
        } catch (e: Throwable) {
            Sentry.captureException(e)
            e.printStackTrace()
        }
        return null;
    }

    @Throws(Throwable::class)
    fun readJson(file: File?): JsonObject {
        val parser = JsonParser()
        var json = parser.parse(FileReader(file)).asJsonObject
        json = json.getAsJsonObject("modules")
        return json
    }

    @Throws(IOException::class)
    fun writeJson(writer: JsonWriter) {
        val modules: MutableList<com.evieclient.modules.Module> = mutableListOf()
        val renderModules: MutableList<RenderModule> = mutableListOf()

        writer.beginObject()
        writer.name("modules")
        writer.beginObject()
        for (module in Evie.MODULE_MANAGER.moduleList) {
            if(modules.contains(module)) continue
            writer.name(module.name)
            writer.value(module.enabled)
            modules.add(module)
        }
        writer.endObject()

        writer.name("render")
        writer.beginObject()
        for (module in Evie.MODULE_MANAGER.renderModuleList) {
            if(renderModules.contains(module)) continue

            writer.name(module.name)
            writer.jsonValue("{\"enabled\": ${module.enabled}, \"x\": ${module.x}, \"y\": ${module.y} }")
            renderModules.add(module)
        }
        writer.endObject()
        writer.endObject()
    }
}